kube-derive/tests/resource_inherit.rs

```diff
@@ -0,0 +1,55 @@
+use k8s_openapi::{
+    api::core::v1::{ConfigMap, Secret},
+    ByteString,
+};
+use kube::api::ObjectMeta;
+use kube_derive::ResourceInherit;
+
+#[derive(ResourceInherit, Default)]
+#[inherit(resource = "ConfigMap")]
+struct TypedMap {
+    metadata: ObjectMeta,
+    data: Option<TypedData>,
+}
```

Comment:
This all looks sensible to me as it stands and happy to include it in the next release. I don't want to hold this up needlessly, but also curious if there are any better suggestions for naming that comes to mind immediately as resource-inherit is a bit of a mouthful.

Is it bad to re-use the trait name `Resource` for the derive proc macro name here? Like, you derive [serde::Deserialize](https://docs.rs/serde/latest/serde/) but it's also a trait separately.

Reply:
I’m for using `Resource` personally, if the name clashes are not a concern. This was the only motivation behind `ResourceInherit`, and the reasoning sound solid to me. 

I was also thinking if the current `CustomResource` can be extended with `inherit(…)` attribute, but didn’t want to make changes to existing established flow.

Also - does it make sense to change example in `errorbounded_configmap_watcher` to utilize the new derive macro?

Reply:
> I’m for using Resource personally

awesome!

> I was also thinking if the current CustomResource can be extended with inherit(…) attribute, but didn’t want to make changes to existing established flow.

uh, possibly. that's probably best served by discussing separately in another thread.

> Also - does it make sense to change example in `errorbounded_configmap_watcher` to utilize the new derive macro?

i think that's an excellent place to show it off.